### PR TITLE
fix: removed onSelectedSectioncChange prop from test details

### PR DESCRIPTION
**Pull request by ayushicopods**
When we are on test Preview we don't need sorting, so we don't need  that prop so  I have make it as optional

### CODE REVIEW

**Review by richa-copods — APPROVED** [review]

**Review by anuragP04 — CHANGES_REQUESTED** [review]

File: app/components/tests/CreateTestPreview.tsx [hunk 1]
```diff
@@ -20,37 +20,28 @@ const TestPreview = ({
...
+    if (
+      (index === 0 && action === "up") ||
+      (index === selectedSections.length - 1 && action === "down")
+    )
+      return
+    else {
+      let temp = selectedSections[index]
+      selectedSections[index] =
+        selectedSections[action == "up" ? index - 1 : index + 1]
+      selectedSections[action == "up" ? index - 1 : index + 1] = temp
```
Comment (Dhruv-Copods) [inline app/components/tests/CreateTestPreview.tsx:39]: Add strict checking

File: app/components/tests/CreateTestPreview.tsx [hunk 2]
```diff
@@ -186,20 +177,28 @@ const TestPreview = ({
                       <div className="flex gap-2">
                         <Icon
                           icon="fa:long-arrow-up"
-                          className="cursor-pointer"
-                          onClick={() => moveSection(i, "up")}
+                          className={`${
+                            i == 0 ? "cursor-not-allowed" : "cursor-pointer"
```
Comment (Dhruv-Copods) [inline app/components/tests/CreateTestPreview.tsx:181]: add strict checking

File: app/components/tests/CreateTestPreview.tsx [hunk 3]
```diff
@@ -186,20 +177,28 @@ const TestPreview = ({
...
+                            if (e.key === "Enter")
+                              changeSelectedSectionsOrder(i, "up")
                           }}
                         />
                         <Icon
                           icon="fa:long-arrow-down"
-                          className="cursor-pointer"
-                          onClick={() => moveSection(i, "down")}
+                          className={`${
+                            i == selectedSections.length - 1
```
Comment (Dhruv-Copods) [inline app/components/tests/CreateTestPreview.tsx:193]: add strict checking

File: app/components/tests/CreateTestPreview.tsx [hunk 4]
```diff
@@ -191,10 +180,11 @@ const TestPreview = ({
                           className={`${
                             i == 0 ? "cursor-not-allowed" : "cursor-pointer"
                           }`}
-                          onClick={() => moveSection(i, "up")}
+                          onClick={() => changeSelectedSectionsOrder(i, "up")}
                           tabIndex={0}
                           onKeyUp={(e) => {
-                            if (e.key === "Enter") moveSection(i, "up")
+                            if (e.key === "Enter")
+                              changeSelectedSectionsOrder(i, "up")
```
Comment (yash6102000) [inline app/components/tests/CreateTestPreview.tsx:187]: can we rename i to index?

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**